odbc/src/api/functions.rs

```diff
@@ -2211,13 +2205,125 @@ unsafe fn sql_get_infow_helper(
+                    i16_len::set_output_fixed_data(
+                        &MONGO_CA2_SUPPORT,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_SCROLL_OPTIONS => {
+                    i16_len::set_output_fixed_data(
+                        &MONGO_SO_SUPPORT,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_BOOKMARK_PERSISTENCE => {
+                    i16_len::set_output_fixed_data(
+                        // We do not support bookmarks
+                        &0,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_NEED_LONG_DATA_LEN => {
+                    $string_func(
+                        SQL_INFO_Y,
+                        info_value_ptr,
+                        buffer_length as usize,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_TXN_ISOLATION_OPTION => {
+                    i16_len::set_output_fixed_data(
+                        &SQL_TXN_SERIALIZABLE,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_DATABASE_NAME => {
+                    let conn = must_be_valid!((*conn_handle).as_connection());
+                    let attributes = conn.attributes.read().unwrap();
+                    if attributes.current_catalog.is_some() {
+                        $string_func(
+                            attributes.current_catalog.as_ref().unwrap().as_str(),
+                            info_value_ptr,
+                            buffer_length as usize,
+                            string_length_ptr,
+                        )
+                    } else {
+                        $string_func(
+                            "test",
+                            info_value_ptr,
+                            buffer_length as usize,
+                            string_length_ptr,
+                        )
+                    }
+                }
+                InfoType::SQL_SCROLL_CONCURRENCY => {
+                    i16_len::set_output_fixed_data(
+                        &SQL_SCCO_READ_ONLY,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_LOCK_TYPES => {
+                    i16_len::set_output_fixed_data(
+                        &SQL_LCK_NO_CHANGE,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_POS_OPERATIONS => {
+                    i16_len::set_output_fixed_data(
+                        &0,
```

Comment:
non-blocking: we group together things that share the same constant value, wondering for all of the `&0`'s, if they are all bitmasks for "unsupported" maybe we can group them together with a comment?

Reply:
I was considering that too. My main reason against that is that they may change independently in the future, but I'm very ambivalent.